influence_benchmark/config/experiment_config.py

```diff
@@ -0,0 +1,154 @@
+from dataclasses import asdict, dataclass, fields
+from typing import Any, Dict, List, Optional, Type, TypeVar, cast
+
+import yaml
+
+from influence_benchmark.config.accelerate_config import AccelerateConfig
+from influence_benchmark.root import EXPERIMENT_CONFIG_DIR
+
+T = TypeVar("T", bound="BaseExperimentConfig")
+
+
+@dataclass
+class BaseExperimentConfig:
+
+    run_name: Optional[str]
+    devices: List[int]
+
+    # Env args
+    env_name: str
+    max_turns: int
+    num_envs_per_device: int
+    max_subenvs_per_env: int
+
+    # Baseiteration args
+    num_gen_trajs_per_initial_state: int
+    top_n_trajs_per_initial_state: int
+    iterations: int
+    log_to_wandb: bool
+    final_reward: bool
+
+    # Training args
+    agent_model_name: str
+    env_model_name: str
+    per_device_train_batch_size: int
+    num_train_epochs: int
+    gradient_accumulation_steps: int
+    gradient_checkpointing: bool
+    learning_rate: float
+    report_to: str
+    optim: str
+    max_seq_length: int
+    lr_scheduler_type: str
+    logging_steps: int
+    lora_r: int
+    lora_alpha: int
+    lora_dropout: float
+
+    # Debugging args
+    seed: Optional[int]
+
+    training_arg_keys = [
+        "agent_model_name",
+        "env_model_name",
+        "per_device_train_batch_size",
+        "num_train_epochs",
+        "gradient_accumulation_steps",
+        "gradient_checkpointing",
+        "learning_rate",
+        "report_to",
+        "optim",
+        "max_seq_length",
+        "lr_scheduler_type",
+        "logging_steps",
+        "lora_r",
+        "lora_alpha",
+        "lora_dropout",
+    ]
+
+    accelerate_config = AccelerateConfig()
+
+    def __post_init__(self):
+        self.accelerate_config.set_gpu_ids(self.devices)
+
+    @classmethod
+    def load(cls: Type[T], config_name: str, devices: Optional[List[int]] = None) -> T:
+        config_path = str(EXPERIMENT_CONFIG_DIR / config_name)
+
+        with open(config_path, "r") as f:
+            config_dict = yaml.safe_load(f)
+
+        if devices is not None:
+            print(f"Overriding GPUs from the config with GPU ids: {devices}")
+            config_dict["devices"] = devices
+
+        return cls.create_config(config_dict)
+
+    @classmethod
+    def create_config(cls: Type[T], config_dict: Dict[str, Any]) -> T:
+        cls._validate_config_keys(config_dict)
```

Comment:
this runs before we set the config type which means that having a beta etc will cause a crash?

Reply:
Ah, thanks for catching this 😬